### PR TITLE
Source Salesforce: concurrent incremental syncs

### PR DESCRIPTION
**Pull request by clnoll**
### What

Updates the Salesforce connector to use the Concurrent CDK for incremental syncs, in addition to full refresh.

### How

Updates the connector to use the new state converter added to the CDK in https://github.com/airbytehq/airbyte/pull/33531.

### CODE REVIEW

File: airbyte-integrations/connectors/source-salesforce/source_salesforce/source.py [hunk 1]
```diff
@@ -50,7 +51,7 @@ class SourceSalesforce(ConcurrentSourceAdapter):
 
     message_repository = InMemoryMessageRepository(Level(AirbyteLogFormatter.level_mapping[logger.level]))
 
-    def __init__(self, catalog: Optional[ConfiguredAirbyteCatalog], config: Optional[Mapping[str, Any]], **kwargs):
+    def __init__(self, catalog: Optional[ConfiguredAirbyteCatalog], config: Optional[Mapping[str, Any]], state: AirbyteStateMessage, **kwargs):
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-salesforce/source_salesforce/source.py:54]: Based on https://github.com/airbytehq/airbyte/pull/33522/files#diff-95988f9ff4d7e4e1a8849df7d7453b3215d82ba5eeba981e7d995ba346973330R24, the input of the methods is `BaseSource.read_state`. This method returns a `TState` and not a `AirbyteStateMessage`

**Review by maxi297 — COMMENTED** [review]
Just one concern but the rest LGTM!

File: airbyte-integrations/connectors/source-salesforce/source_salesforce/source.py (hunk 1 above)
Comment (clnoll) [reply airbyte-integrations/connectors/source-salesforce/source_salesforce/source.py:54]: 👍 Thanks, fixed.

**Review by girarda — APPROVED** [review]
looks great! :shipit:

**Review by maxi297 — APPROVED** [review]
:shipit:

**Pull request merged**